File: cglib/src/main/java/com/chogoon/cglib/CgObservable.java
package com.chogoon.cglib;

import androidx.databinding.ObservableArrayList;

public interface CgObservable<T> {

    ObservableArrayList getObservableArrayList();

    void getData(Object data);

    void onData(T data);

    void submit(Object data);

    void onComplete();

    void onClear();

    void isNoData(boolean value);


}

File: cglib/src/main/java/com/chogoon/cglib/CgObservable.kt
package com.chogoon.cglib

import androidx.databinding.ObservableArrayList

interface CgObservable<T> {

    fun getObservableArrayList(): ObservableArrayList<*>

    fun getData(any: Any?)
    fun onData(data: T)
    fun submit(any: Any?)
    fun onComplete()
    fun onClear()
    fun isNoData(flag: Boolean)

}